### DISCUSSION TITLE
audio2csv won't write events to a csv

### DISCUSSION THREAD

**Discussion by DougPHynes** [QutEcoacoustics/audio-analysis#520] [Q&A] [answered]
Firstly, thanks to the team for making Analysis Programs available to all!

My problem concerns the generic recognizer. I can’t seem to get the audio2csv function to write acoustic events (e.g., "EventStartSeconds") to a csv when using my own recognizer config file (I can get the program to write events when using the "Towsey.LewiniaPectoralis.yml" config file). My configuration file appears to do a good job of recognizing events (i.e., it draws boxes around the Blackpoll Warbler songs, as seen in the sonograms), but the “Ecosounds.GenericRecognizer.Events.csv” file comes up empty (my files and logs are attached).

I appreciate any insights you might have! 

[RecognizerTest.zip](https://github.com/QutEcoacoustics/audio-analysis/files/7434051/RecognizerTest.zip)

**Comment by atruskie** [comment, marked as answer]
Hi @DougPHynes,

We're trialing a new format for the generic recognisers - they can recognise more complex events.

Look for the `events.beta.csv` file?

![image](https://user-images.githubusercontent.com/1452968/139605541-3e775ce7-3636-4b38-bf55-7d1fb424029f.png)

Now, admittedly, the new format has some clear problems, but the events are there.

For the missing columns in that output, we can file an actual issue to backfill them.

**Reply by DougPHynes** [reply]
Thanks very much for the reply @atruskie. Yes, I noticed the data within the "beta" csv–those are great.

A little background on my thoughts and what I'm trying (or thinking) to do. I was intrigued by the variables suggested within the headers of the “Ecosounds.GenericRecognizer.Events" csv; namely the start and end times, and low and high frequencies, variables which seem to correspond to a bounding box for each event. While I'm not exactly sure how I would use such bounding boxes (or labels), it'd seem I could import them into other software to extract more acoustic parameters to for eventual machine learning, as noted here: 
https://ap.qut.ecoacoustics.info/guides/generic_recognizers.html#10-building-a-larger-data-set

Do you think this is a valid approach? I'm keen to continue to try to use AP to identify calls/songs/events because I found it did a great job at recognizing the songs in my wav file.

**Reply by atruskie** [reply]
Yeah a totally valid approach. 

Those extra columns _are missing_ from the new CSV output. I just went to make an actual bug report and found my old one: https://github.com/QutEcoacoustics/audio-analysis/issues/506 

I've got some cycles allocated for a month, last half of November to work on AP. I'll get on to it.

**Reply by atruskie** [reply]
@DougPHynes I forgot to note that all this information is in the _.json_ file. Both Excel and R can read JSON files.

**Reply by DougPHynes** [reply]
Great tip! I've just read the .json file into R and successfully retrieved the event start and end times.
